### PR TITLE
add doc page about retry-ability

### LINKED ISSUE #1278: Document retry-ability
This is a key concept, yet we don't have a good single documentation place for it (something that on.cypress.io/retry-ability could refer to). So far we have two places

- https://docs.cypress.io/faq/questions/using-cypress-faq.html#How-can-I-make-Cypress-wait-until-something-is-visible-in-the-DOM
- https://docs.cypress.io/guides/core-concepts/introduction-to-cypress.html#How-Cypress-really-looks-Promises-wrapped-up-and-hidden-from-us where it is mentioned in passing when talking about Promises

We need a section in the docs specifically about it, and how assertions that follow are retried, and even how to add it to your own custom commands and maybe even using `cypress-pipe`

https://glebbahmutov.com/blog/cypress-should-callback/#custom-commands-with-retry

### PR DESCRIPTION
**Pull request by bahmutov**
## done
- [x] link to `cypress-pipe` and retrying clicks from blog post https://www.cypress.io/blog/2019/01/22/when-can-the-test-click/
- closes #1278 

## items below will be done separately

I am thinking how to better address these, since they don't really fit into the example used through the page. It might take another separate PR 

- [ ] explain how to implement retries in custom commands, use cypress-xpath as an example
- [ ] address example and question from https://github.com/cypress-io/cypress/issues/3184

Code example in https://github.com/cypress-io/cypress-example-todomvc/tree/doc-retry-ability

### CODE REVIEW

**Comment by jennifer-shehane** [comment]
You want this reviewed now? I see some checkboxes not done, so I'm unsure.

**Comment by bahmutov** [comment]
@jennifer-shehane I think the unchecked boxes should be addressed in separate PRs. For example, adding retry-ability to the custom commands should be documented in the custom commands guide. so I would like to land this so I can create a section in the testing workshop based on this doc.

File: source/guides/guides/retry-ability.md [hunk 1]
```diff
@@ -2,41 +2,55 @@
...
 
-Retry-ability is the core part of Cypress Test Runner that allows it to test dynamic web applications. Like a good car transmission, it usually works without you noticing it, but understanding how it works will help you write faster tests with fewer run-time surprises.
+{% note info %}
+# {% fa fa-graduation-cap %} What you'll learn
+- How Cypress retries commands and assertions
+- When commands are retried and when they are not
+- How to address some situations of flaky tests
+{% endnote %}
+
+A core feature of Cypress that assists with testing dynamic web applications is retry-ability. Like a good transmission in a car, it usually works without you noticing it. But understanding how it works will help you write faster tests with fewer run-time surprises.
```
Comment (bahmutov) [inline source/guides/guides/retry-ability.md]: thank you for leaving in my car transmission analogy!

Comment (jennifer-shehane) [reply source/guides/guides/retry-ability.md]: It is good! 😆

**Comment by jennifer-shehane** [comment]
Dependent on this: https://github.com/cypress-io/cypress-monorepo/issues/221

**Comment by bahmutov** [comment]
oh no @jennifer-shehane 

```
1) Guides Table of Contents displays toc:
     CypressError: cy.request() failed on:

http://localhost:2222/guides/core-concepts/retry-ability.html

The response we received from your web server was:

  > 404: Not Found

This was considered a failure because the status code was not '2xx' or '3xx'.

If you do not want status codes to cause failures pass the option: 'failOnStatusCode: false'

-----------------------------------------------------------

The request we sent was:

Method: GET
URL: http://localhost:2222/guides/core-concepts/retry-ability.html
Headers: {
  "user-agent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Cypress/3.1.5 Chrome/59.0.3071.115 Electron/1.8.2 Safari/537.36",
  "accept": "*/*",
  "accept-encoding": "gzip, deflate"
}
```

**Comment by jennifer-shehane** [comment]
oh, duh, I forgot to move the actual file 🙄 Will fix now

**Comment by bahmutov** [comment]
Relevant testing workshop section https://gitpitch.com/cypress-io/testing-workshop-cypress?p=slides/11-retry-ability

**Pull request merged**